src/prop/cadical.cpp

```diff
@@ -204,6 +204,33 @@ unsigned CadicalSolver::getAssertionLevel() const
 
 bool CadicalSolver::ok() const { return d_inSatMode; }
 
+void CadicalSolver::setDrat()
+{
+  std::ostringstream filePathStringStream;
+  filePathStringStream << "dratProof_" << getpid() << ".drat";
+  d_tempDratFilePath = filePathStringStream.str();
+  d_dratFile = fopen(d_tempDratFilePath.c_str(), "wb");
+  // Currently only non-binary proofs are supported
+  d_solver->set_long_option("--no-binary");
+  d_solver->trace_proof(d_dratFile, d_tempDratFilePath.c_str());
+}
+
+std::ifstream CadicalSolver::getDrat()
+{
+  fclose(d_dratFile);
+  d_solver->close_proof_trace();
+  std::ifstream dratFile(d_tempDratFilePath, std::ios::binary);
+  return dratFile;
+}
+
+void CadicalSolver::deleteDratFile()
+{
+  if (!d_tempDratFilePath.empty())
+  {
+    std::remove(d_tempDratFilePath.c_str());
```

Comment:
```suggestion
    std::filesystem::remove(d_tempDratFilePath.c_str());
```

Reply:
This should be `std::filesystem::remove` since `std::remove` does not delete a file.

Reply:
It was working like this 🤔 
But I updated it anyway, thanks